src/main/kotlin/com/sourcegraph/cody/config/CodyAccount.kt

```diff
@@ -22,14 +25,34 @@ data class CodyAccount(
     @Attribute("id") override var id: String = generateId(),
 ) : ServerAccount() {
 
-  fun getAccountType(): AccountType {
-    if (isDotcomAccount()) {
-      return AccountType.DOTCOM
-    }
-    return AccountType.ENTERPRISE
+  @Volatile private var isProUser: Boolean? = null
+
+  fun getAccountType(project: Project): String {
```

Comment:
I don't like that change. The difference b/w enterprise and free/pro is different than the difference b/w free and pro. We can call the first category "account type" and the second a "tier". And my feeling is that we are adjusting the model to the telemetry requirements (which feel bad).

If we really need to consider enterprise/free/pro as a same category maybe let's introduce an enum for them. Is `AccountType` still needed or should we deprecate it? 

Also, if we do the agent call here the method name should be altered to reflect the hidden process - "derive" instead of "get"?

Reply:
I can move this code the the logger as it's not user anywhere else so far.